stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java

```diff
@@ -759,6 +759,9 @@ public TypeSpec getTypeAdapterSpec(@NotNull TypeTokenConstantsGenerator typeToke
         }
 
         AnnotatedClass annotatedClass = SupportedTypesModel.getInstance().getSupportedType(typeMirror);
+        if(null == annotatedClass) {
+            throw new IllegalStateException("The AnnotatedClass class can't be null in TypeAdapterGenerator : " + typeMirror.toString());
+        }
```

Comment:
The function getSupportedType can be called from anywhere and it can throw null based on the state whether all the classes have been added or not. Also it is an independent class which does not really have the information in which order it can be called.

The reason why I added an IllegalStateException here is because the SupportedTypesModel should have been populated with this class type before calling this function.